src/plugins/syscalls/syscalls.cpp

```diff
@@ -205,155 +220,51 @@ static uint64_t transform_value(drakvuf_t drakvuf, drakvuf_trap_info_t* info, co
-        keyval("Module", fmt::Qstr(std::move(module))),
-        keyval("vCPU", fmt::Nval(info->vcpu)),
-        keyval("CR3", fmt::Xval(info->regs->cr3)),
-        keyval("Syscall", fmt::Nval(nr)),
-        keyval("Ret", fmt::Nval(ret)),
-        keyval("Info", fmt::Rstr(extra_info ?: ""))
-    );
-}
-
-static GHashTable* read_syscalls_filter(const char* filter_file)
-{
-    GHashTable* table = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, NULL);
-    if (!table) return NULL;
-
-    FILE* f = fopen(filter_file, "r");
-    if (!f)
+    std::ifstream file(filter_file);
+    if (!file.is_open())
     {
-        g_hash_table_destroy(table);
-        return NULL;
+        fprintf(stderr, "[SYSCALLS] failed to open syscalls file, does it exist?\n");
+        return false;
     }
-    ssize_t bytes_read;
-    do
-    {
-        char* line = NULL;
-        size_t len = 0;
-        bytes_read = getline(&line, &len, f);
-        while (bytes_read > 0 && (line[bytes_read - 1] == '\n' || line[bytes_read - 1] == '\r')) bytes_read--;
-        if (bytes_read > 0)
-        {
-            line[bytes_read] = '\0';
-            g_hash_table_insert(table, line, NULL);
-        }
-        else
-            free(line);
-    } while (bytes_read != -1);
 
-    fclose(f);
-    return table;
-}
-
-void free_trap(gpointer p)
-{
-    if ( !p )
-        return;
-
-    drakvuf_trap_t* t = (drakvuf_trap_t*)p;
-    if ( t->data )
-        g_slice_free(struct wrapper, t->data);
+    std::string line;
+    while (std::getline(file, line))
+    {
+        line = rstrip(std::move(line));
+        if (!line.empty())
+            this->filter.insert(line);
+    }
 
-    g_slice_free(drakvuf_trap_t, t);
+    return true;
 }
 
-}
 
-syscalls::syscalls(drakvuf_t drakvuf, const syscalls_config* c, output_format_t output)
-    : pluginex(drakvuf, output)
-    , traps(NULL)
-    , strings_to_free(NULL)
-    , filter(NULL)
-    , format{output}
-    , offsets(NULL)
-    , win32k_profile{c->win32k_profile ?: ""}
+syscalls_base::syscalls_base(drakvuf_t drakvuf, const syscalls_config* config, output_format_t output) : pluginex(drakvuf, output)
 {
     this->os = drakvuf_get_os_type(drakvuf);
     this->kernel_base = drakvuf_get_kernel_base(drakvuf);
-    this->reg_size = drakvuf_get_address_width(drakvuf); // 4 or 8 (bytes)
+    this->register_size = drakvuf_get_address_width(drakvuf); // 4 or 8 (bytes)
     this->is32bit = (drakvuf_get_page_mode(drakvuf) != VMI_PM_IA32E);
-    this->disable_sysret = c->disable_sysret;
+    this->disable_sysret = config->disable_sysret;
 
-    if ( c->syscalls_filter_file )
-        this->filter = read_syscalls_filter(c->syscalls_filter_file);
-
-    if ( this->os == VMI_OS_WINDOWS )
-        setup_windows(drakvuf, this, c);
-    else
-        setup_linux(drakvuf, this);
+    if (config->syscalls_filter_file && !this->read_syscalls_filter(config->syscalls_filter_file))
+        throw -1;
 }
 
-syscalls::~syscalls()
```

Comment:
You need to keep the destructor, otherwise you get the memleaks spotted by Valgrind